ui-admin/src/portal/publish/diffComponents.tsx

```diff
@@ -24,77 +24,169 @@ export const VersionChangeView = ({ record }: {record: VersionedEntityChange}) =
   </div>
 }
 
+type ConfigChangesProps = {
+  configChanges: ConfigChange[],
+  selectedChanges: ConfigChange[],
+  updateSelectedChanges: (changes: ConfigChange[]) => void
+}
 /** renders a list of config changes, or "no changes" if empty */
-export const ConfigChanges = ({ configChanges }: {configChanges: ConfigChange[]}) => {
+export const ConfigChanges = ({ configChanges, selectedChanges, updateSelectedChanges }: ConfigChangesProps) => {
   if (!configChanges.length) {
     return <span className="fst-italic text-muted">no changes</span>
   }
+  /** handles add/remove from the selected changes */
+  const updateSelection = (propertyName: string, selected: boolean) => {
+    const selectionIndex = selectedChanges.findIndex(change => change.propertyName === propertyName)
+    if (selected && selectionIndex < 0) {
+      updateSelectedChanges([
+        ...selectedChanges,
+        configChanges.find(change => change.propertyName === propertyName) as ConfigChange
```

Comment:
I think we should update ESLint configuration to allow non-null assertions and use them in cases like this. I think that would make it clearer that this line is assuming something is not null vs casting another type to ConfigChange. 

https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#non-null-assertion-operator-postfix-